app/src/main/java/com/postpc/tenq/services/RecorderService.java

```diff
@@ -0,0 +1,138 @@
+package com.postpc.tenq.services;
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.media.MediaRecorder;
+import android.util.Log;
+import android.os.Handler;
+
+import static java.lang.Math.abs;
+import static java.lang.Math.exp;
+
+public class RecorderService {
+
+    private static double mEMA = 0.0;
+    static final private double EMA_FILTER = 0.6;
+
+    private MediaRecorder mRecorder;
+    private final Context applicationContext;
+    private static Thread runner;
+    private boolean isRecorderOn = true; // TODO: change to false when not debugging
+    private double prevAmplitude;
+    private AudioManager audioManager;
+
+    public RecorderService(Context applicationContext) {
+
+        this.applicationContext = applicationContext;
+        this.audioManager = (AudioManager) this.applicationContext.getSystemService(Context.AUDIO_SERVICE);
+        this.prevAmplitude = getAmplitude();
+
+        if (runner == null && isRecorderOn)
+        {
+            runner = new Thread(() -> {
+                while (runner != null)
+                {
+                    try
+                    {
+                        Thread.sleep(6000);
+                    }
+                    catch (InterruptedException e) {
+                        android.util.Log.e("[Monkey]", "InterruptedException: " + android.util.Log.getStackTraceString(e));
+                    }
+
+                    double currentAmplitude = getAmplitude();
+                    int maxVolume = audioManager.getStreamMaxVolume(AudioManager.STREAM_MUSIC);
+                    Log.i("amplitude", String.valueOf(currentAmplitude));
```

Comment:
I think it's best not to push debug log lines to common code - it will end up in production :)

Reply:
Is there a way to print only when being in debug mode?

Reply:
I don't know, but you can skip these lines when you commit your files / revert these changes